### PR TITLE
Account for current and expanded tables in tryRehash

### PR DESCRIPTION
**Pull request by arhimondr**
## Description

While rehash is in process both current and expanded hash tables are retained

## Release notes

(X) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java [hunk 1]
```diff
@@ -275,12 +275,11 @@ private boolean tryRehash()
 
         // An estimate of how much extra memory is needed before we can go ahead and expand the hash table.
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java:276]: the comment above is no longer consistent

File: core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java [hunk 2]
```diff
@@ -312,6 +311,8 @@ private boolean tryRehash()
         groupIds = newGroupIds;
 
         this.valuesByGroupId = Arrays.copyOf(valuesByGroupId, maxFill);
+
+        preallocatedMemoryInBytes = 0;
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java:315]: did you forget to call `updateMemory.update()` after this?

**Comment by arhimondr** [comment]
Updated

File: core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java [hunk 3]
```diff
@@ -312,6 +311,9 @@ private boolean tryRehash()
...
+        updateMemory.update();
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/BigintGroupByHash.java:316]: add a comment why the second update (releasing tmp mem)

File: core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java [hunk 4]
```diff
@@ -394,6 +393,9 @@ private boolean tryRehash()
         this.maxFill = calculateMaxFill(newCapacity);
         this.rawHashByHashPosition = rawHashes;
         this.groupIdsByHash = newGroupIdByHash;
+
+        preallocatedMemoryInBytes = 0;
+        updateMemory.update();
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java:398]: same as above

**Review by sopel39 — APPROVED** [review]

**Pull request merged**